### PR TITLE
Add test for driver-only HMAC

### LINKED ISSUE #8564: Add test for driver-only HMAC
Add a test component in `all.sh` with `MD_C` disabled (make sure there's no `mbedtls_md_hmac` function in `md.o`) and `KEY_TYPE_HMAC`, `ALG_HMAC` as well as all hashes accelerated. Also a reference component and parity testing in `analyze_outcomes.py`.

These component would ideally use the full config, however restrictions might be necessary (for example, TLS CBC functions use HMAC and it's not clear whether they will work with driver-only HMAC). This task's goal is is not to extend support for driver-only HMAC, merely to clarify what level of support we already have. (However if we identify a quick win, we might as well take it.)

### PR DESCRIPTION
**Pull request by valeriosetti**
## Description

This PR add a couple of test components to `all.sh`:
- `component_test_psa_crypto_config_accel_hmac` which accelerates HMAC and all hash alorithms
- `component_test_psa_crypto_config_reference_hmac` which is used as reference component

Test coverage parity is ensured by adding the proper task to `analyze_outcomes.py`.

Resolves #8564 

## PR checklist

- [x] **changelog** not required
- [x] **backport** not required
- [x] **tests** provided

### CODE REVIEW

File: tests/scripts/all.sh [hunk 1]
```diff
@@ -3655,6 +3655,68 @@ component_test_psa_crypto_config_reference_hash_use_psa() {
...
+                    ALG_SHA3_224 ALG_SHA3_256 ALG_SHA3_384 ALG_SHA3_512"
+
+    # Configure
+    # ---------
+
+    config_psa_crypto_hmac_use_psa 1
+
+    # Disable MD_C in order to disable the builtin support for HMAC. MD_LIGHT
+    # is still enabled though.
+    scripts/config.py unset MBEDTLS_MD_C
```
Comment (mpg) [inline tests/scripts/all.sh:3690]: Shouldn't this go in `config_psa_crypto_hmac_use_psa()`? Currently that function accepts a parameter (`driver_only="$1"`) and then does nothing with it.

Comment (mpg) [reply tests/scripts/all.sh:3690]: I think we should also disable all built-in hashes as a matter of principle. We declared them as supported by the driver, so there's no reason we shouldn't be able to remove the built-ins. (And if there's one that I'm missing, I want to know.)

File: tests/scripts/all.sh [hunk 2]
```diff
@@ -3655,6 +3655,68 @@ component_test_psa_crypto_config_reference_hash_use_psa() {
...
+                    ALG_SHA_224 ALG_SHA_256 ALG_SHA_384 ALG_SHA_512 \
+                    ALG_SHA3_224 ALG_SHA3_256 ALG_SHA3_384 ALG_SHA3_512"
+
+    # Configure
+    # ---------
+
+    config_psa_crypto_hmac_use_psa 1
+
+    # Disable MD_C in order to disable the builtin support for HMAC. MD_LIGHT
+    # is still enabled though.
```
Comment (mpg) [inline tests/scripts/all.sh:3689]: Suggestion: "is still enabled though (for ENTROPY_C among others)".

**Review by mpg — COMMENTED** [review]
Going in the right direction. In addition to the specific comments below, could you also update `docs/driver-only-build.md` with a new section about HMAC? (Sorry, should have been in the task's description.)

File: tests/scripts/analyze_outcomes.py [hunk 3]
```diff
@@ -246,6 +246,9 @@ def do_analyze_driver_vs_reference(results: Results, outcomes: Outcomes, args) -
             'component_ref': 'test_psa_crypto_config_reference_hmac',
             'component_driver': 'test_psa_crypto_config_accel_hmac',
             'ignored_suites': [
+                # These suites require legacy hash support, which is disabled
+                # in the accelerate component.
```
Comment (mpg) [inline tests/scripts/analyze_outcomes.py:250]: Typo: accelerateD

File: docs/driver-only-builds.md [hunk 4]
```diff
@@ -105,7 +105,26 @@ provided by a driver or built-in, you should use the following macros:
...
+In addition to accelerated hash operations, it is also possible to accelerate
+HMAC by enabling and accelerating:
+- HMAC algorithm and key type, i.e. `[PSA_WANT|MBEDTLS_PSA_ACCEL]_ALG_HMAC` and
+  `[PSA_WANT|MBEDTLS_PSA_ACCEL]KEY_TYPE_HMAC`.
+- Required hash algorithm(s) as explained in [Hashes](#hashes) section.
+
+In such a build it is possible to disable legacy HMAC support by disabling
+`MBEDTLS_MD_C` and still getting crypto operations, X.509 and TLS to work as
+usual. Exceptions are:
+- [Hashes](#hashes) are obviously valid here for the accelerated algorithms.
```
Comment (mpg) [inline docs/driver-only-builds.md:121]: I'm not sure I understand this sentence, sorry. Do you mean the restrictions mentioned there apply here too, or something else?

**Review by mpg — APPROVED** [review]
LGTM, thanks!

File: tests/suites/test_suite_entropy.function [hunk 5]
```diff
@@ -447,7 +447,7 @@ void entropy_nv_seed_std_io()
 }
 /* END_CASE */
 
-/* BEGIN_CASE depends_on:MBEDTLS_MD_C:MBEDTLS_ENTROPY_NV_SEED:MBEDTLS_PLATFORM_NV_SEED_ALT */
+/* BEGIN_CASE depends_on:MBEDTLS_MD_LIGHT:MBEDTLS_ENTROPY_NV_SEED:MBEDTLS_PLATFORM_NV_SEED_ALT */
```
Comment (minosgalanakis) [inline tests/suites/test_suite_entropy.function:450]: What happens if `defined(MBEDTLS_MD_LIGHT) && !defined(MBEDTLS_MD_C)` ? Do we want the `entropy_nv_seed` test case to be run?

Comment (valeriosetti) [reply tests/suites/test_suite_entropy.function:450]: It seems to me that both `entropy_nv_seed()` in `test_suite_entropy` and functions from `entropy.c` depend only on `MD_LIGHT` functions so this change should be fine. Am I missing something?

**Comment by gilles-peskine-arm** [comment]
@minosgalanakis Do I correctly understand that you're reviewing this PR? (In which case I won't review it, and please remove the "needs-reviewer" label.)

**Comment by minosgalanakis** [comment]
> @minosgalanakis Do I correctly understand that you're reviewing this PR? (In which case I won't review it, and please remove the "needs-reviewer" label.)

Yes sorry that I missed the flag. I was about to review it just wanted a clarification on a dependency edge case

**Review by minosgalanakis — APPROVED** [review]
LGTM

**Pull request merged**